File: DSA/DSA06039 SỐ ĐẦU TIÊN BỊ LẶP.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    vector<int> A(n);
    for (auto &&i : A) cin >> i;

    unordered_set<int> check;
    int ans = n;

    for (int i = n-1; i >= 0; i--)
    {
        if (check.count(A[i]))
        {
            if (ans > i) ans = i;
        }
        check.insert(A[i]);
    }
    
    ans != n ? cout << A[ans] : cout << "NO";


    cout << endl;
}


int main()
{
    ios::sync_with_stdio(false);                                    // Speed down input/output
    cin.tie(nullptr);
    cout.tie(nullptr);
    
    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();
    
    while (so_bo_test--)
    {
        submit();
    }

    return 0;
}

File: DSA_python/DSA06039 SỐ ĐẦU TIÊN BỊ LẶP.py
def submit():
    n = int(input())
    A = list(map(int, input().split()))

    check = set()
    ans = n
    for i in range(n-1, -1, -1):
        if A[i] in check:
            if ans > i: ans = i
        check.add(A[i])
    
    print(A[ans] if ans != n else "NO")

test = int(input())
for i in range(test):
    submit()
